### ISSUE TITLE
Slices are empty even though they should not be

### ISSUE THREAD

**Issue by Markus-Goetz** [helmholtz-analytics/heat#592]
**Description**
Receiving empty tensor after slicing even though it should contain elements. 

**To Reproduce**
Please see the following script.

[Script](https://github.com/helmholtz-analytics/heat/blob/features/556-assign_label/examples/classification/demo_knn.py
)

Index list is created in line 87. I am try to slice my data (here: x) using this index list in l. 91. The resulting tensor should contain elements, however, it is empty.

**Version Info**
master

**Comment by coquelin77** [comment]
this happens because in the master advanced indexing was not implemented properly. This should be fixed in PR #594

**Comment by coquelin77** [comment]
PS to do blocks of code in markdown use three of those tick marks, i.e. ` ` ` (without the spaces). You can also give the language after the first set, i.e. ` ` ` python ...

**Comment by coquelin77** [comment]
upon testing there are a couple things to correct in your script and a few other things to know about HEAT itself

1. using the `_DNDarray__array` property of the DNDarray will return the local PyTorch Tensor. Changes to the local tensor are invisible to the global DNDarray. If changes are made on the local tensor it can cause other heat functions to malfunction.
2. the `array` function is intended for creating a new DNDarray from existing data, if the data is already split between processes then you need to use the `is_split` parameter instead of the `split` parameter.
3. you do not need to print the torch tensor, the print on the heat array also works.

```python
import heat as ht

indices = [0, 2]
data = ht.array([1, 2, 3, 4], split=0)

print(data._DNDarray__array, data.comm.rank)

print(data[indices])
result = ht.array(data[indices], split=0)
print(result._DNDarray__array, result.comm.rank)
```
results in 
```
[0] tensor([1, 2]) 0
[0] tensor([1])
[0] tensor([1]) 0
[1] tensor([3, 4]) 1
[1] tensor([3])
[1] tensor([], dtype=torch.int64) 1
```
the last result is empty because the `array` function is given the wrong parameter

**Comment by Markus-Goetz** [comment]
Should be resolved in the master now by PR #594
